code/__HELPERS/lists.dm

```diff
@@ -344,3 +344,13 @@
 	while(L.Remove(null))
 		continue
 	return L
+
+//Copies a list, and all lists inside it recusively
+//Does not copy any other reference type
+/proc/deepCopyList(list/l)
+	if(!islist(l))
+		return l
+	. = l.Copy()
+	for(var/i = 1 to length(.))

```

Comment:
I find the ```length()``` op on lists ugly, because like, they have a variable -exactly for this kind of thing- and it's VASTLY more common to see ```list.len``` than it is ```length(list)``` in our code.